### ISSUE TITLE
[RnD] Passkey support

### ISSUE THREAD

**Issue by teon** [DefGuard/defguard#399]
How we should implement it?

**Comment by wojcik91** [comment]
@teon Regarding support for usernameless login (similar to GitHub):
- requires the use of passkeys supporting [Discoverable Credentials / Resident Keys](https://developers.yubico.com/WebAuthn/WebAuthn_Developer_Guide/Resident_Keys.html)
- our already registered passkeys don't have this enabled
- the crate we use for webauthn stuff has this feature in theory
- the authors discourage using this feature (https://github.com/kanidm/webauthn-rs/issues/388) since it can soft-brick some security keys 
- due to this the implementation is also a WIP with prominent TODOs: https://github.com/kanidm/webauthn-rs/blob/master/webauthn-rs/src/lib.rs#L1421

In summary due to our security focus I'd probably stick to using usernames for now and see how things shake out at a later date.
